extensions/vertx/runtime/src/main/java/io/quarkus/vertx/runtime/VertxRecorder.java

```diff
@@ -69,24 +72,48 @@ void destroy() {
     void registerMessageConsumers(Map<String, ConsumeEvent> messageConsumerConfigurations) {
         if (!messageConsumerConfigurations.isEmpty()) {
             EventBus eventBus = vertx.eventBus();
+            VertxInternal vi = (VertxInternal) VertxRecorder.vertx;
             CountDownLatch latch = new CountDownLatch(messageConsumerConfigurations.size());
-            final List<Throwable> registrationFailures = new ArrayList();
+            final List<Throwable> registrationFailures = new ArrayList<>();
             for (Entry<String, ConsumeEvent> entry : messageConsumerConfigurations.entrySet()) {
                 EventConsumerInvoker invoker = createInvoker(entry.getKey());
                 String address = entry.getValue().value();
-                MessageConsumer<Object> consumer;
-                if (entry.getValue().local()) {
-                    consumer = eventBus.localConsumer(address);
-                } else {
-                    consumer = eventBus.consumer(address);
-                }
-                consumer.handler(new Handler<Message<Object>>() {
+                // Create a context attached to each consumer
+                // If we don't all consumers will use the same event loop and so published messages (dispatched to all
+                // consumers) delivery is serialized.
+                Context context = vi.createEventLoopContext();
```

Comment:
So IIUIC without this change all message consumers would be executed on the same event loop thread, right? And it's related to this change: https://vert-x3.github.io/vertx-4-migration-guide/index.html#_context_affinity_for_non_vert_x_thread.

So the `createEventLoopContext()` basically "distributes" the consumers to all available event loops?